src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs

```diff
@@ -351,6 +349,7 @@ public void OnEmbeddedGUI()
                 var isPublished = Repository.CurrentRemote.HasValue;
                 if (isPublished)
                 {
+                    var currentRemote = Repository.CurrentRemote.Value.Name;
```

Comment:
We actually want to cache both `isPublished` and `currentRemote` in serialized fields and update them in `MaybeUpdateData` so that the UI doesn't get confused if this information changes in a background thread while the UI is in the middle of a layout/repaint phase.